src/main/java/frc/robot/subsystems/MAXSwerveModule.java

```diff
@@ -125,9 +135,77 @@ public MAXSwerveModule(int driveCANID, int steerCANID, double chassisAngularOffs
     this.m_steerSparkMax.setIdleMode(SwerveModuleConstants.kSteerMotorIdleMode);
     this.m_driveSparkMax.setSmartCurrentLimit(SwerveModuleConstants.kDriveMotorCurrentLimit);
     this.m_steerSparkMax.setSmartCurrentLimit(SwerveModuleConstants.kSteerMotorCurrentLimit);
+
+    // Burn the configurations. If a SparkMax browns out during operation, it will maintain the above
+    // configurations
+    this.m_driveSparkMax.burnFlash();
+    this.m_steerSparkMax.burnFlash();
+
+    // Set chassis offset, the current module state angle position of the steer encoder, and zeros the drive encoder
+    this.m_chassisAngularOffset = chassisAngularOffset;
+    this.m_desiredState.angle = new Rotation2d(this.m_steerEncoder.getPosition());
+    this.m_driveEncoder.setPosition(0);
+
+    // Simulation setup
+    if (RobotBase.isSimulation()) {
+        REVPhysicsSim.getInstance().addSparkMax(this.m_driveSparkMax, DCMotor.getNEO(1));
+        REVPhysicsSim.getInstance().addSparkMax(this.m_steerSparkMax, DCMotor.getNeo550(1));
+    }
+  }
+
+  // Returning Values
+  public double getDrivePosition() {
+    return this.m_driveEncoder.getPosition();
+  }
+
+  public double getDriveVelocity() {
+    return this.m_driveEncoder.getVelocity();
+  }
+
+  public double getHeadingDegrees() {
+    return this.m_steerEncoder.getPosition();
+  }
+
+  public Rotation2d getHeading() {
+    return Rotation2d.fromDegrees(this.getHeadingDegrees());
+  }
+
+  // Returning Swerve State
+  public SwerveModuleState getState() {
+    return new SwerveModuleState(this.getDriveVelocity(), this.getHeading());
+  }
+  // Returning Swerve Position
+  public SwerveModulePosition getPosition() {
+    return new SwerveModulePosition(this.getDrivePosition(), this.getHeading());
+  }
+  
+  public void resetDriveEncoder() {
```

Comment:
Methods like this don't technically need to contain javadocs since the name is descriptive enough. However, just to get used to making these docs, we'll also need one here!